File: src/main/java/com/dingdongdeng/coinautotrading/common/type/CoinExchangeType.java
package com.dingdongdeng.coinautotrading.common.type;

import java.util.Arrays;
import java.util.EnumMap;
import java.util.NoSuchElementException;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum CoinExchangeType {
    UPBIT("업비트"),
    ;

    private String desc;

    public static CoinExchangeType of(String name) {
        return Arrays.stream(CoinExchangeType.values())
            .filter(type -> type.name().equalsIgnoreCase(name))
            .findFirst()
            .orElseThrow(() -> new NoSuchElementException(name));
    }

    public static EnumMap<CoinExchangeType, String> toMap() {
        EnumMap<CoinExchangeType, String> map = new EnumMap<>(CoinExchangeType.class);
        for (CoinExchangeType value : CoinExchangeType.values()) {
            map.put(value, value.getDesc());
        }
        return map;
    }
}

File: application-infra/common/src/main/kotlin/com/dingdongdeng/autotrading/type/CoinExchangeType.kt
package com.dingdongdeng.autotrading.type

import java.util.EnumMap

enum class CoinExchangeType(
    val desc: String,
    val marketType: MarketType,
) {
    UPBIT("업비트", MarketType.SPOT),
    BINANCE_FUTURE("바이낸스 선물", MarketType.FUTURE);

    companion object {
        fun of(name: String?): CoinExchangeType {
            return values().first { type -> type.name.equals(name, ignoreCase = true) }
        }

        fun toMap(): EnumMap<CoinExchangeType, String> {
            val map = EnumMap<CoinExchangeType, String>(
                CoinExchangeType::class.java
            )
            for (value in values()) {
                map[value] = value.desc
            }
            return map
        }
    }
}
